src/types.ts

```diff
@@ -10,6 +10,8 @@ declare global {
 export type WindType = 'true' | 'false' | 'speed' | 'direction' | 'barb' | 'barb-and-speed' | 'barb-and-direction' | 'barb-speed-and-direction';
 export type ShowDateType = 'false' | 'boundary' | 'all';
 export type IconFillType = 'single' | 'full' | number;
+export type Conditions = 'clear-night' | 'cloudy' | 'fog' | 'hail' | 'lightning' | 'lightning-rainy' | 'partlycloudy' | 'pouring' | 'rainy' | 'snowy' | 'snowy-rainy' | 'sunny' | 'windy' | 'windy-variant' | 'exceptional'
+export type IconMap = Partial<Record<Conditions, string>>
```

Comment:
```suggestion
export type Condition = 'clear-night' | 'cloudy' | 'fog' | 'hail' | 'lightning' | 'lightning-rainy' | 'partlycloudy' | 'pouring' | 'rainy' | 'snowy' | 'snowy-rainy' | 'sunny' | 'windy' | 'windy-variant' | 'exceptional';
type PerConditionConfig<TValue> = Partial<Record<Condition, TValue>>;
export type IconMap = PerRecordConfig<string>;
```

Reply:
And then replace the `ColorConfig` interface below with:

```
export type ColorConfig extends PerConditionConfig<ColorDefinition>;
```

Reply:
And edit the type of `condition` in `ConditionSpan` and `ForecastSegment` to `Condition` instead of `string`.

Reply:
I couldn't make it work with `export type ColorConfig extends PerConditionConfig<ColorDefinition>;`. I tried `export interface ColorConfig extends PerConditionConfig<ColorDefinition> {};` but then I got a bunch of type errors every time it iterates over ColorConfig key/values.

I'm not sure if what I did is equivalent tbh.

Reply:
Oh, my bad - I originally had that comment suggesting an interface, but changed to a type and forgot to adjust the syntax. You got it right.